src/vcpkg/export.prefab.cpp

```diff
@@ -19,22 +19,20 @@ namespace vcpkg::Export::Prefab
 
     static std::vector<path> find_modules(const VcpkgPaths& system, const path& root, const std::string& ext)
     {
-        std::vector<path> paths;
-        Filesystem& utils = system.get_filesystem();
-        std::error_code error_code;
-        if (!utils.exists(root, error_code) || !utils.is_directory(root)) return paths;
-
-        stdfs::recursive_directory_iterator it(root);
-        stdfs::recursive_directory_iterator endit;
-
-        while (it != endit)
+        Filesystem& fs = system.get_filesystem();
+        std::error_code ec;
+        auto paths = fs.get_regular_files_recursive(root, ec);
+        if (ec && !(ec == std::errc::not_a_directory || ec == std::errc::no_such_file_or_directory))
```

Comment:
Please double check this part closely, I don't know exactly what export.prefab does.

Reply:
```suggestion
        if (ec == std::errc::not_a_directory || ec == std::errc::no_such_file_or_directory) return paths;
        if (ec)
```
I think this is slightly nicer.

Reply:
Is the error code from `get_regular_files_recursive` always w.r.t. the root or can it be due to failures during the recursive travel? The original code only gives a pass for the root not existing.

Reply:
>I think this is slightly nicer.

I did the bool check first because the "are you that ec" check is much more expensive. I can try to separate out from that if.

Reply:
> Is the error code from `get_regular_files_recursive` always w.r.t. the root or can it be due to failures during the recursive travel?

Pretty sure from anywhere but I wouldn't expect either of those error codes except for on the root.

>The original code only gives a pass for the root not existing.

I'm pretty sure the original code would crash if there were errors elsewhere in the tree because it uses ++.